File: backend_programming/c++/labs/exact_change.cpp
#include <iostream>
using namespace std;

int main() {

    int money;

    int dollars = 100;
    int quarters = 25;
    int dimes = 10;
    int nickels = 5;

    int dollarAmount;
    int quarterAmount;
    int dimeAmount;
    int nickelAmount;

    int dollarLeftover;
    int quarterLeftover;
    int dimeLeftover;
    int nickelLeftover;

    cin >> money;



    if (money)
    {
        //DOLLARS
        dollarAmount = money / dollars; //this will group the dollars together
        dollarLeftover = money % dollars; //this will find the remainders that can't be grouped by 100s
        if (dollarAmount > 0)
        {
            if (dollarAmount > 1)
            {
                cout << dollarAmount << " Dollars" << endl;
            }
            else
            {
                cout << dollarAmount << " Dollar" << endl;
            }
        }

        //QUARTERS
        quarterAmount = dollarLeftover / quarters; //this will group the quarters together
        quarterLeftover = dollarLeftover % quarters; //this will find the remainders that can't be grouped by 25s
        if (quarterAmount > 0)
        {
            if (quarterAmount > 1)
            {
                cout << quarterAmount << " Quarters" << endl;
            }
            else
            {
                cout << quarterAmount << " Quarter" << endl;
            }
        }


        //DIMES
        dimeAmount = quarterLeftover / dimes; //this will group dimes together
        dimeLeftover = quarterLeftover % dimes; //this will find the remainders that can't be grouped by 10s
        if (dimeAmount > 0)
        {
            if (dimeAmount > 1)
            {
                cout << dimeAmount << " Dimes" << endl;
            }
            else
            {
                cout << dimeAmount << " Dime" << endl;
            }
        }


        //NICKELS
        nickelAmount = dimeLeftover / nickels; //this will group nickels together
        nickelLeftover = dimeLeftover % nickels; //this will find the remainders that can't be grouped by 5s
        if (nickelAmount > 0)
        {
            if (nickelAmount > 1)
            {
                cout << nickelAmount << " Nickels" << endl;
            }
            else
            {
                cout << nickelAmount << " Nickel" << endl;
            }
        }


        //PENNIES
        if (nickelLeftover > 0)
        {
            if (nickelLeftover > 1)
            {
                cout << nickelLeftover << " Pennies" << endl;
            }
            else
            {
                cout << nickelLeftover << " Penny" << endl;
            }
        }
    }
    else if (money == 0)
    {
        cout << "No change" << endl;
    }

    return 0;
}

File: backend_programming/java/WGU_Labs/exactChange.java
import java.util.Scanner; 

public class LabProgram {
   public static void main(String[] args) {
      /* Type your code here. */
      
      Scanner scnr = new Scanner(System.in);
      int change = scnr.nextInt();
      
      if (change == 0)
      {
         System.out.println("No change");   
      }
      else
      {
         int dollars;
         int quarters;
         int dimes;
         int nickels;
         int pennies;
         
         dollars = change / 100;
         change = change - (dollars * 100);
         if (dollars > 1)
         {
            System.out.printf("%d Dollars%n", dollars);   
         }
         else if (dollars == 1)
         {
            System.out.printf("%d Dollar%n", dollars);   
         }
         
         quarters = change / 25;
         change = change - (quarters * 25);
         if (quarters > 1)
         {
            System.out.printf("%d Quarters%n", quarters);   
         }
         else if (quarters == 1)
         {
            System.out.printf("%d Quarter%n", quarters);   
         }
         
         dimes = change / 10;
         change = change - (dimes * 10);
         if (dimes > 1)
         {
            System.out.printf("%d Dimes%n", dimes);   
         }
         else if (dimes == 1)
         {
            System.out.printf("%d Dime%n", dimes);   
         }
         
         nickels = change / 5;
         change = change - (nickels * 5);
         if (nickels > 1)
         {
            System.out.printf("%d Nickels%n", nickels);   
         }
         else if (nickels == 1)
         {
            System.out.printf("%d Nickel%n", nickels);   
         }
         
         pennies = change / 1;
         change = change - (pennies * 1);
         if (pennies > 1)
         {
            System.out.printf("%d Pennies%n", pennies);   
         }
         else if (pennies == 1)
         {
            System.out.printf("%d Penny%n", pennies);   
         }
      }
   }
}
